docs/rfc/rfc-6.adoc

```diff
@@ -0,0 +1,228 @@
+
+We will implement a simple reimbursement contract for maintainers proxying calls
+to Bridge functions. Only approved maintainers should be able to use the proxy
+contract. Maintainers are expected to follow round-robin distribution of work
+but the order is not going to be enforced on-chain. DAO/council will have the
+functionality of removing maintainers not performing their duties.
+DAO/council may decide to distribute additional T rewards for maintainers. This
+functionality should be supported by the Maintainer contract.
+
+Eventually, the role of maintainers can be delegated to another network such as
+Gelato.
+
+
+```
+┌────────────────────────────────────────┐   calls  ┌─────────────────────────┐
+|               Maintainer               |─────────▶|          Bridge         |
+└────────────────────────────────────────┘          └─────────────────────────┘  
+| submitSweeProof() onlyMaintainer       |          | submitSweepProof()      | 
+| submitRedemptionProof() onlyMaintainer |          | submitRedemptionProof() | 
+| startDkg() onlyMaintainer              |          | startDkg()              | 
+└────────────────────────────────────────┘          └─────────────────────────┘
+```
+
+===== Submit sweep proof
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+===== Submit redemption proof
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+===== Start DKG
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers. 
+
+===== Report DKG timeout
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+==== Third party incentives
+
+
+All transactions in this section must be supported by the off-chain client. We
+are not concerned about front-running. Front-running may happen and it is even
+desirable to some extent because these transactions protect the health of the
+network.
+
+[[challenge-dkg-result]]
+===== Challenge DKG result
+
+No ETH cost reimbursed. Called exceptionally, ideally never. Incentivized by T
+from slashed stakes.
+
+===== Notify redemption timeout
+
+
+No ETH cost reimbursed. Called exceptionally, ideally never. Incentivized by T
+from slashed stakes.
+
+[[report-fraud]]
+===== Report fraud
+
+No ETH cost reimbursed. Called exceptionally, ideally never. Incentivized by T
+from slashed stakes.
+
+=== ETH pool
+
+DAO needs to fund ETH pool that will be used for reimbursements. The pool should
+probably be a separate contract. The pool needs to protect against malicious
+miner-operators by placing a governable gas price ceiling. It should be possible
+to withdraw unspent ETH in case we decide to replace Maintainer incentives with
+something else (for example Gelato) or decide to add more functions there
+(for example, the SPV relay updates). 
+
+== Open questions
+
+- Should the DAO/council approve the wallet's decisions to mark some members as
```

Comment:
Ineligibility for rewards will result in the rewards coming for this particular wallet or all rewards in a specific period of time?

The majority sounded good to me at first. If we think that having 51 votes is not enough we could raise the limit to e.g. 64 votes.

But on the other hand what if there is an upgrade of the client that will result in the majority of members staying on the old version accusing members that updated the version. Or when some part of the members is using Ethereum API provider that is down. Having a safety would be wise. Maybe instead of approving the majority decision the DAO could reject the decision and bring a member back to eligible for rewards? Accepting all the decisions may be a PITA.

Reply:
> Ineligibility for rewards will result in the rewards coming for this particular wallet or all rewards in a specific period of time?

A particular operator is marked as ineligible for future rewards, for example, for the next 2 weeks.

> The majority sounded good to me at first. If we think that having 51 votes is not enough we could raise the limit to e.g. 64 votes.

We also need to balance the cost of validating signatures. Something we can decide during the implementation.

> But on the other hand what if there is an upgrade of the client that will result in the majority of members staying on the old version accusing members that updated the version. Or when some part of the members is using Ethereum API provider that is down. Having a safety would be wise. Maybe instead of approving the majority decision the DAO could reject the decision and bring a member back to eligible for rewards? Accepting all the decisions may be a PITA.

I think that the DAO should be able to turn rewards back on for someone. But it should not approve decisions of the group every single time. We could think about it for slashing but not for rewards ineligibility IMO.